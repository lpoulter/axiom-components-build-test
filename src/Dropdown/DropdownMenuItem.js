import PropTypes from 'prop-types';
import React, { Component } from 'react';
import omit from 'lodash.omit';
import ContextMenuItem from '../Context/ContextMenuItem';
import DropdownReactContext from './DropdownReactContext';

export default class DropdownMenuItem extends Component {
  static propTypes = {
    /** SKIP */
    children: PropTypes.node,
    /** Disabled interactions and applies styling */
    disabled: PropTypes.bool,
   /**
    * Whether the menu should stay open after being clicked.
    */
    keepOpen: PropTypes.bool,
     /**
     * Whether the menu is part of a multi-selection menu. When true, this
     * will disable automatically closing the DropdownMenu when clicked.
     */
    multiSelect: PropTypes.bool,
    /** Click handler  */
    onClick: PropTypes.func,
    /**  Vertical padding size applied to the menu item */
    paddingVertical: PropTypes.oneOf(['small', 'medium']),
    /** Provides indication that the menu item is selected */
    selected: PropTypes.bool,
  };

  static contextType = DropdownReactContext;

  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(event) {
    const { closeDropdown } = this.context;
    const { multiSelect, keepOpen, onClick } = this.props;

    if (closeDropdown && !multiSelect && !keepOpen) closeDropdown();

    if (onClick) onClick(event);
  }

  render() {
    const { children, ...rest } = this.props;

    return (
      <ContextMenuItem
          { ...omit(rest, ['keepOpen']) }
          onClick={ this.handleClick }
          tabIndex="0">
        { children }
      </ContextMenuItem>
    );
  }
}
