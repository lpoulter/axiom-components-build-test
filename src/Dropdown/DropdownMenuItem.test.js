import React from 'react';
import { shallow } from 'enzyme';
import DropdownMenuItem from './DropdownMenuItem';
import toJson from 'enzyme-to-json';

const render = (props = {}, opts = {}) => shallow(
  <DropdownMenuItem { ...props }>
    Test
  </DropdownMenuItem>, opts);

describe('DropdownMenuItem', () => {
  let props;
  let opts;
  let mockEvent;

  beforeEach(() => {
    props = {
      disabled: false,
      keepOpen: false,
      multiSelect: false,
      onClick: jest.fn(),
    };
    opts = {
      context: {
        closeDropdown: jest.fn(),
      },
    };
    mockEvent = {};
  });

  describe('onClick', () => {
    it('closes the dropdown menu when clicked', () => {
      render(props, opts).simulate('click', mockEvent);
      expect(opts.context.closeDropdown).toHaveBeenCalledTimes(1);
    });

    it('calls the click property with the event when clicked', () => {
      render(props, opts).simulate('click', mockEvent);
      expect(props.onClick).toHaveBeenCalledTimes(1);
      expect(props.onClick).toHaveBeenCalledWith(mockEvent);
    });

    it('does not close the dropdown when multiSelect is set', () => {
      props.multiSelect = true;
      render(props, opts).simulate('click', mockEvent);
      expect(opts.context.closeDropdown).not.toHaveBeenCalled();
    });

    it('does not close the dropdown when keepOpen is set', () => {
      props.keepOpen = true;
      render(props, opts).simulate('click', mockEvent);
      expect(opts.context.closeDropdown).not.toHaveBeenCalled();
    });
  });

  describe('Snapshots', () => {
    it('renders with defaultProps', () => {
      expect(toJson(render({}, opts))).toMatchSnapshot();
    });

    it('renders with disabled', () => {
      expect(toJson(render({
        disabled: true,
      }, opts))).toMatchSnapshot();
    });

    it('renders with index', () => {
      expect(toJson(render({
        index: '1',
      }, opts))).toMatchSnapshot();
    });

    it('renders with keepOpen', () => {
      expect(toJson(render({
        keepOpen: true,
      }, opts))).toMatchSnapshot();
    });

    it('renders with multiSelect', () => {
      expect(toJson(render({
        multiSelect: true,
      }, opts))).toMatchSnapshot();
    });

    it('renders with onClick', () => {
      expect(toJson(render({
        onClick: () => {},
      }, opts))).toMatchSnapshot();
    });

    it('renders with onMouseOver', () => {
      expect(toJson(render({
        onMouseOver: () => { },
      }, opts))).toMatchSnapshot();
    });
  });
});
