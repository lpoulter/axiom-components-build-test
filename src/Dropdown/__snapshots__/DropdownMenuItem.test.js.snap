// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenuItem Snapshots renders with defaultProps 1`] = `
<ContextMenuItem
  onClick={[Function]}
  paddingVertical="medium"
  tabIndex="0"
>
  Test
</ContextMenuItem>
`;

exports[`DropdownMenuItem Snapshots renders with disabled 1`] = `
<ContextMenuItem
  disabled={true}
  onClick={[Function]}
  paddingVertical="medium"
  tabIndex="0"
>
  Test
</ContextMenuItem>
`;

exports[`DropdownMenuItem Snapshots renders with index 1`] = `
<ContextMenuItem
  index="1"
  onClick={[Function]}
  paddingVertical="medium"
  tabIndex="0"
>
  Test
</ContextMenuItem>
`;

exports[`DropdownMenuItem Snapshots renders with keepOpen 1`] = `
<ContextMenuItem
  onClick={[Function]}
  paddingVertical="medium"
  tabIndex="0"
>
  Test
</ContextMenuItem>
`;

exports[`DropdownMenuItem Snapshots renders with multiSelect 1`] = `
<ContextMenuItem
  multiSelect={true}
  onClick={[Function]}
  paddingVertical="medium"
  tabIndex="0"
>
  Test
</ContextMenuItem>
`;

exports[`DropdownMenuItem Snapshots renders with onClick 1`] = `
<ContextMenuItem
  onClick={[Function]}
  paddingVertical="medium"
  tabIndex="0"
>
  Test
</ContextMenuItem>
`;

exports[`DropdownMenuItem Snapshots renders with onMouseOver 1`] = `
<ContextMenuItem
  onClick={[Function]}
  onMouseOver={[Function]}
  paddingVertical="medium"
  tabIndex="0"
>
  Test
</ContextMenuItem>
`;
